### PR TITLE
Multiple metrics reporters

### PR DESCRIPTION
**Pull request by tsegismont**
Supersedes #60 from @dankristensen

### CODE REVIEW

**Comment by dankristensen** [comment]
What has happened to this issue. I did not see it in 3.5.0 as you wrote earlier?

File: reporters/influxdb/README.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+= vertx-metric-reporters-influxdb
+:source-language: java
+
+image::https://vertx.ci.cloudbees.com/buildStatus/icon?job=vert.x3-hawkular-metrics[link=https://vertx.ci.cloudbees.com/job/vert.x3-hawkular-metrics/]
+
+Vert.x-Metrics-reporters-influxdb contains metrics collection and reporting to InfluxDb.
+
+Hawkular is an http://www.influxdb.org[open source monitoring and management solution].
```
Comment (jotak) [inline reporters/influxdb/README.adoc:8]: s/Hawkular/Influx

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+ * |{@code vertx.net.server.<host>:<port>.connections}
+ * |Number of opened connections to the Net Server listening on the {@code <host>:<port>} address.
+ *
+ * |Counter
+ * |{@code vertx.net.server.<host>:<port>.bytesReceived}
+ * |Total number of bytes received by the Net Server listening on the {@code <host>:<port>} address.
+ *
+ * |Counter
+ * |{@code vertx.net.server.<host>:<port>.bytesSent}
+ * |Total number of bytes sent to the Net Server listening on the {@code <host>:<port>} address.
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:171]: I guess it's "number of bytes sent **by** the Net Server"
(same applies in Hawkular doc)

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+ * |{@code vertx.http.server.<host>:<port>.connections}
+ * |Number of opened connections to the HTTP Server listening on the {@code <host>:<port>} address.
+ *
+ * |Counter
+ * |{@code vertx.http.server.<host>:<port>.bytesReceived}
+ * |Total number of bytes received by the HTTP Server listening on the {@code <host>:<port>} address.
+ *
+ * |Counter
+ * |{@code vertx.http.server.<host>:<port>.bytesSent}
+ * |Total number of bytes sent to the HTTP Server listening on the {@code <host>:<port>} address.
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:197]: I guess it's "number of bytes sent **by** the HTTP Server"
(same applies in Hawkular doc)

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+ *
+ * |Counter
+ * |{@code vertx.http.server.<host>:<port>.errorCount}
+ * |Total number of errors.
+ *
+ * |Gauge
+ * |{@code vertx.http.client.<host>:<port>.requests}
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:204]: s/client/server
(same applies in Hawkular doc)

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+ * |Number of requests being processed.
+ *
+ * |Counter
+ * |{@code vertx.http.client.<host>:<port>.requestCount}
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:208]: s/client/server

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+ * |Total number of requests processed.
+ *
+ * |Counter
+ * |{@code vertx.http.client.<host>:<port>.processingTime}
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:212]: s/client/server

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+ * |Cumulated request processing time.
+ *
+ * |Gauge
+ * |{@code vertx.http.client.<host>:<port>.wsConnections}
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:216]: s/client/server

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+ * |Counter
+ * |{@code vertx.pool.<type>.<name>.delay}
+ * |Cumulated time waiting for a resource (queue time).
+ *
+ * |Gauge
+ * |{@code vertx.pool.<type>.<name>.queued}
+ * |Current number of elements waiting for a resource.
+ *
+ * |Counter
+ * |{@code vertx.pool.<type>.<name>.queueCount}
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:344]: 'd' missing: queue**d**Count

File: reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+ * |Counter
+ * |{@code vertx.pool.<type>.<name>.completed}
+ * |Total number of elements done with the resource (i.e. total number of tasks executed for worker pools).
+ *
+ * |Gauge
+ * |{@code vertx.pool.<type>.<name>.maxPoolSize}
+ * |Maximum pool size, only present if it could be determined.
+ *
+ * |Gauge
+ * |{@code vertx.pool.<type>.<name>.inUse}
```
Comment (jotak) [inline reporters/influxdb/src/main/java/io/vertx/ext/influxdb/package-info.java:364]: `vertx.pool.<type>.<name>.poolRatio`

**Review by jotak — COMMENTED** [review]
Some doc issues (both in existing hawkular and influx), apart from that it looks good to me.

**Comment by jotak** [comment]
@dankristensen , we have been working on rearchitecturing the different vertx-metrics-xxx repositories.
Now there is vertx-monitoring: https://github.com/vert-x3/vertx-monitoring
The first PR ( https://github.com/vert-x3/vertx-monitoring/pull/1 ) contains your work and @tsegismont 's,  and is planned to be released in 3.5.1

**Comment by dankristensen** [comment]
@jotak, that sounds fantastic. I have been awaiting this.

**Comment by tsegismont** [comment]
Closing as there is now `vertx-micrometer` which allows to have multiple backends.

**Pull request closed without merging**